spec/octokit/client_spec.rb

```diff
@@ -422,6 +422,134 @@
     end
   end
 
+  describe "redirect handling" do
+    it "follows redirect for 301 response" do
+      client = oauth_client
+
+      original_request = stub_get("/foo").
+        to_return(:status => 301, :headers => { "Location" => "/bar" })
+      redirect_request = stub_get("/bar").to_return(:status => 200)
+
+      client.get("/foo")
+      assert_requested original_request
+      assert_requested redirect_request
+    end
+
+    it "follows redirect for 302 response" do
+      client = oauth_client
+
+      original_request = stub_get("/foo").
+        to_return(:status => 302, :headers => { "Location" => "/bar" })
+      redirect_request = stub_get("/bar").to_return(:status => 200)
+
+      client.get("/foo")
+      assert_requested original_request
+      assert_requested redirect_request
+    end
+
+    it "follows redirect for 307 response" do
+      client = oauth_client
+
+      original_request = stub_post(github_url("/foo")).
+        with(:body => { :some_property => "some_value" }.to_json).
+        to_return(:status => 307, :headers => { "Location" => "/bar" })
+      redirect_request = stub_post(github_url("/bar")).
+        with(:body => { :some_property => "some_value" }.to_json).
+        to_return(:status => 201, :headers => { "Location" => "/bar" })
+
+      client.post("/foo", { :some_property => "some_value" })
+      assert_requested original_request
+      assert_requested redirect_request
+    end
+
+    it "follows redirects for supported HTTP methods" do
+      client = oauth_client
+
+      http_methods = [:head, :get, :post, :put, :patch, :delete]
+
+      http_methods.each do |http|
+        original_request = stub_request(http, github_url("/foo")).
+          to_return(:status => 301, :headers => { "Location" => "/bar" })
+        redirect_request = stub_request(http, github_url("/bar")).
+          to_return(:status => 200)
+
+        client.send(http, "/foo")
+        assert_requested original_request
+        assert_requested redirect_request
+      end
+    end
+
+    it "does not change HTTP method when following a redirect" do
+      client = oauth_client
+
+      original_request = stub_get("/foo").
+        to_return(:status => 301, :headers => { "Location" => "/bar" })
+      redirect_request = stub_get("/bar").to_return(:status => 200)
+
+      client.get("/foo")
```

Comment:
If an HTTP client was going to change the verb when following the redirect, it's most likely to change the verb to a `GET`. To increase this test's ability to catch future regressions, I recommend using something other than a `GET` for this test (e.g., maybe use a `DELETE`).